internal/types/error.go

```diff
@@ -16,6 +16,9 @@ const (
 	ErrorUnknownTOMLKey               ErrorCode = "unknownTOMLKey"
 	ErrorInvalidConfigFiles           ErrorCode = "invalidConfigFiles"
 	ErrorCredentialServiceUnavailable ErrorCode = "credentialsServiceUnavailable"
+	ErrorDeploymentTargetNotFound     ErrorCode = "deploymentTargetNotFound"
+	ErrorDeploymentTargetIsForbidden  ErrorCode = "deploymentTargetIsForbidden"
+	ErrorDeploymentTargetUnreachable  ErrorCode = "deploymentTargetUnreachable"
```

Comment:
Do we need three unique `deploymentTarget` error types here? Or is it sufficient to pass this information within the error itself? The lines of code are free, but the more types we have the more names we have to keep track of + write in all the places, etc.

Reply:
The concept was that these were three types of errors, which we needed to pass through within the error, so that the UX could identify them and place them into a specific error type (to be handled according per the type). Having them listed in one place allows us to keep them synchronized from one place, since we're not sharing code between the agent and the UX. 

I believe this concept still holds and is a good practice, since the error code here is functioning as a const or enum.